lib/credentials/ecs_credentials.js

```diff
@@ -69,23 +79,18 @@ AWS.ECSCredentials = AWS.util.inherit(AWS.Credentials, {
    */
   request: function request(path, callback) {
     path = path || '/';
-
-    var data = '';
-    var http = AWS.HttpClient.getInstance();
     var httpRequest = new AWS.HttpRequest('http://' + this.host + path);
     httpRequest.method = 'GET';
     httpRequest.headers.Accept = 'application/json';
-    var httpOptions = this.httpOptions;
-
-    process.nextTick(function() {
-      http.handleRequest(httpRequest, httpOptions, function(httpResponse) {
-        httpResponse.on('data', function(chunk) { data += chunk.toString(); });
-        httpResponse.on('end', function() { callback(null, data); });
-      }, callback);
-    });
+    AWS.util.handleRequestWithTimeoutRetries(httpRequest, this, callback);
   },
 
   /**
+   * @api private
+   */
+  refreshQueue: [],
```

Comment:
EC2MetdataCredentials already had this feature, through the MetadataService. The `refreshQueue` on the ECSCredentials is analogous to the `loadCredentialsCallbacks` on the MetadataService. They're structured slightly differently but work the same way. (`refreshQueue` has to retain references to the instances of ECSCredentials because the data isn't given to the callback the way it is MetadataService).